File: Bloom/Bloom/Models/GeneratedSession.swift
import Foundation

struct GeneratedSession: Identifiable, Decodable, Hashable {
    let id: UUID
    let user_id: UUID
    let source_voice_profile_id: UUID?
    let title: String
    let subtitle: String?
    let script_text: String
    let audio_url: String
    let generation_status: String
    let generation_provider: String?
    let provider_job_id: String?
    let session_type: String
    let duration_seconds: Int?
    let cover_emoji: String?
    let created_at: String
    let updated_at: String
}

struct GenerateSessionResponse: Decodable {
    let success: Bool
    let generated_session: GeneratedSession
}

File: Bloom/bloom-backend/app/schemas/generated_session.py
from pydantic import BaseModel
from uuid import UUID
from typing import Optional


class GeneratedSessionResponse(BaseModel):
    id: UUID
    user_id: UUID
    source_voice_profile_id: Optional[UUID] = None
    title: str
    subtitle: Optional[str] = None
    script_text: str
    audio_url: str
    generation_status: str
    generation_provider: Optional[str] = None
    provider_job_id: Optional[str] = None
    session_type: str
    duration_seconds: Optional[int] = None
    cover_emoji: Optional[str] = None
    created_at: str
    updated_at: str
